rmw_connext_xtypes_dynamic_cpp/src/functions.cpp

```diff
@@ -202,7 +202,8 @@ rmw_create_publisher(const rmw_node_t * node,
 
     DDSDomainParticipant* participant = (DDSDomainParticipant*)node->data;
 
-    if (type_support->typesupport_identifier != rosidl_typesupport_introspection_cpp::typesupport_introspection_identifier)
+    // TODO avoid string comparison
+    if (strcmp(type_support->typesupport_identifier, rosidl_typesupport_introspection_cpp::typesupport_introspection_identifier))
```

Comment:
What would you suggest to pass as `n`? Because if I use `strlen` to determine the length the problem would be same if the string wouldn't have a terminating zero character.